modules/kubernetes-addons/aws-ebs-csi-driver/main.tf

```diff
@@ -33,7 +36,7 @@ module "irsa_addon" {
 }
 
 resource "aws_iam_policy" "aws_ebs_csi_driver" {
-  count = local.create_irsa ? 1 : 0
+  count = local.create_irsa && var.enable_amazon_eks_aws_ebs_csi_driver || var.enable_self_managed_aws_ebs_csi_driver ? 1 : 0
```

Comment:
likewise - this should stay as it was since IRSA is valid for either route

Reply:
to deploy the addon in self_managed mode i use the terraform module "helm_addon" which contains [a block for IRSA management](https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/896/files#diff-a3e7919ec85e9b3b1439fe0505889c99852929bcb828868f7545c2bb5870a0f6R82-R88). so i used it and disabled this resource when we are in self managed mode

Reply:
```suggestion
  count = local.create_irsa || var.enable_self_managed_aws_ebs_csi_driver ? 1 : 0
```